pokedex_async_redux/lib/features/widget/pokemon_overview_page_gridview.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:pokedex_async_redux/api/model/pokemon.dart';
+import 'package:pokedex_async_redux/utilities/constants.dart';
+import 'package:flutter/material.dart';
+
+class PokemonOverviewGridview extends StatelessWidget {
+  const PokemonOverviewGridview({
+    super.key,
+    required this.pokemons,
+  });
+
+  final List<Pokemon> pokemons;
+
+  @override
+  Widget build(BuildContext context) {
+    return GridView.builder(
+      gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: numberTwo),
+      itemCount: pokemons.length,
+      itemBuilder: (context, index) {
+        final pokemon = pokemons[index];
+        final pokemonIndex = index + numberOne;
+        final String pokemonImageUrlPath =
+            pokemonImageUri.replaceAll(RegExp(r'\[index\]'), '$pokemonIndex') + pokemonImageExtention;
```

Comment:
u don't need regex here, a simple string manipulation will do, instead u might use that for getting pokemon id. look at jayvee's pr for reference on she implemented this part 🙂